pkg/kubectl/scale.go

```diff
@@ -184,14 +201,87 @@ type realScalerClient struct {
 	client client.Interface
 }
 
+// GetReplicationController returns the replication controller with the provided namespace/name
 func (c *realScalerClient) GetReplicationController(namespace, name string) (*api.ReplicationController, error) {
 	return c.client.ReplicationControllers(namespace).Get(name)
 }
 
+// UpdateReplicationController updates the provided replication controller in the given namespace
 func (c *realScalerClient) UpdateReplicationController(namespace string, rc *api.ReplicationController) (*api.ReplicationController, error) {
 	return c.client.ReplicationControllers(namespace).Update(rc)
 }
 
-func (c *realScalerClient) ControllerHasDesiredReplicas(rc *api.ReplicationController) wait.ConditionFunc {
-	return client.ControllerHasDesiredReplicas(c.client, rc)
+// WaitForReplicas is a closure around the WaitForReplicas function
+func (c *realScalerClient) WaitForReplicas(rc *api.ReplicationController, delta int, timeout time.Duration) error {
+	return WaitForReplicas(c.client, rc, delta, timeout)
+}
+
+// WaitForReplicas accepts a replication controller and waits until it either observes delta pod events
+// (additions, deletions) or times out
+func WaitForReplicas(client client.Interface, rc *api.ReplicationController, delta int, timeout time.Duration) error {
+	done := make(chan struct{})
+	stopCh := make(chan struct{})
+	changes := 0
+
+	switch {
```

Comment:
previously we had `ControllerHasDesiredReplicas` which took an rc and waited till spec.Replcas == status.Replicas. You should be able to mirror those semantics by waiting till len(store.List()) == spec.Replicas and avoid this delta thing?